src/cryptoadvance/specter/tor_daemon.py

```diff
@@ -65,13 +65,24 @@ def get_hashed_password(self, password):
         return hashed_pw
 
     def is_running(self):
-        # This fails the cypress-tests, unfortunately
-        # if not self.tor_daemon_proc and self.is_port_open():
-        #    raise SpecterError(
-        #        "Port 9050 is open but tor_daemon_proc is not existing. Probably another Tor-Daemon is running?!"
-        #    )
-        return self.tor_daemon_proc and self.tor_daemon_proc.poll() is None
+        """Checks whether the Tor process is still running.
+        Note: poll() from the subprocess module does not work reliably. For example,
+        if the process has been terminated but its exit code has not yet been collected, it would still indicate
+        that the process is still running.
+        """
```

Comment:
I'm not that convinced about these arguments. Are there open issues we can refer to? Or StackOverflow discussions?
What is meant by: "if the process has been terminated but its exit code has not yet been collected, it would still indicate".
The [API-docs](https://docs.python.org/3/library/subprocess.html#subprocess.Popen) read like:
"Popen.poll() : Check if child process has terminated. Set and return [returncode](https://docs.python.org/3/library/subprocess.html#subprocess.Popen.returncode) attribute. Otherwise, returns None."

Reply:
Well, the open issue was, that it didn't work reliably for me during testing and the above works like a charm.
"If the process has been terminated but its exit code has not yet been collected, it would still indicate ... " means that the poll method would indicate in this instance that the process is still running when in reality it is not (which was the case during testing).